### PR TITLE
Modify deals_with_adjustments_for_deleted_source method of adjustment…

### PR DESCRIPTION
**Pull request by nishant-cyro**
…_source, add two scopes in adjustment and write their specs

### CODE REVIEW

File: core/app/models/spree/adjustment.rb [hunk 1]
```diff
@@ -67,6 +67,8 @@ class Adjustment < Spree::Base
     scope :is_included, -> { where(included: true) }
     scope :additional, -> { where(included: false) }
     scope :competing_promos, -> { where(source_type: competing_promos_source_types) }
+    scope :with_complete_order, -> { joins(:order).merge(Spree::Order.where.not(completed_at: nil)) }
```
Comment (priyank-gupta) [inline core/app/models/spree/adjustment.rb]: @nishant-cyro I think we should name this scope as `for_complete_order`.

File: core/app/models/spree/adjustment.rb [hunk 2]
```diff
@@ -67,6 +67,8 @@ class Adjustment < Spree::Base
...
+    scope :with_incomplete_order, -> { joins(:order).merge(Spree::Order.where(completed_at: nil)) }
```
Comment (priyank-gupta) [inline core/app/models/spree/adjustment.rb]: @nishant-cyro I think we should name this scope as `for_incomplete_order`.

File: core/app/models/spree/adjustment.rb (hunk 1 above)
Comment (nishant-cyro) [inline core/app/models/spree/adjustment.rb]: @priyank-gupta - done

File: core/app/models/spree/adjustment.rb (hunk 2 above)
Comment (nishant-cyro) [inline core/app/models/spree/adjustment.rb]: @priyank-gupta done!

File: core/lib/spree/core/validators/db_maximum_length_validator.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class DbMaximumLengthValidator < ActiveModel::Validator
```
Comment (priyank-gupta) [inline core/lib/spree/core/validators/db_maximum_length_validator.rb]: @nishant-cyro Not sure why this custom validator is in this PR ?

Comment (nishant-cyro) [inline core/lib/spree/core/validators/db_maximum_length_validator.rb]: @priyank-gupta - removed.

**Comment by priyank-gupta** [comment]
:+1: 
Create a direct PR and then close this.

**Pull request closed without merging**